src/pages/workspace/distanceRates/PolicyNewDistanceRatePage.tsx

```diff
@@ -0,0 +1,111 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import AmountForm from '@components/AmountForm';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapperWithRef from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import getPermittedDecimalSeparator from '@libs/getPermittedDecimalSeparator';
+import * as MoneyRequestUtils from '@libs/MoneyRequestUtils';
+import * as NumberUtils from '@libs/NumberUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import {createPolicyDistanceRate, generateCustomUnitID} from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/PolicyCreateDistanceRateForm';
+import type {Rate} from '@src/types/onyx/Policy';
+import type Policy from '@src/types/onyx/Policy';
+
+type PolicyNewDistanceRatePageOnyxProps = {
+    policy: OnyxEntry<Policy>;
+};
+
+type PolicyDistanceRatePageProps = PolicyNewDistanceRatePageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.CREATE_DISTANCE_RATE>;
+
+function PolicyNewDistanceRatePage({policy, route}: PolicyDistanceRatePageProps) {
+    const styles = useThemeStyles();
+    const {translate, toLocaleDigit} = useLocalize();
+    const currency = policy !== null ? policy?.outputCurrency : CONST.CURRENCY.USD;
+    const customUnits = policy?.customUnits ?? {};
+    const customUnitID = customUnits[Object.keys(customUnits)[0]].customUnitID;
```

Comment:
Will this throw if `customUnits` is an empty object?

Reply:
Seems like it does
![Screenshot 2024-03-12 at 4 34 52 PM](https://github.com/Expensify/App/assets/22219519/48774456-f510-4e3b-9ecf-43895d27520d)